tokio-macros/src/entry.rs

```diff
@@ -325,6 +325,7 @@ fn parse_knobs(
     let brace_token = input.block.brace_token;
     input.block = syn::parse2(quote_spanned! {last_stmt_end_span=>
         {
+            #[allow(clippy::expect_used)]
```

Comment:
Note that this `#[allow]` also applied to the `#body`. That's not necessarily the behavior that users of `clippy::expect_used` want.

Reply:
Oh, that's a good point :)

Reply:
One solution I was thinking about was to convert it to panic!, but that is just moving the problem elsewhere (And quick search proved to me that `#[deny(clippy::panic)] lints are used in the wild). 

I tried to find a way to generate a better output without any panics and have an implementation that returns a generic error trait from the generated main() function (and passing the async closure error to it, if the #body closure is supposed to return one)

The problem is, that it brings out some other issues, like failing #[should_panic] tokio::test units, that apparently can't have a return value :(

Reply:
Should I go with the panic! version? (Dunno if it's really an acceptable solution), or try to make additional exceptions for tests?

Reply:
How about separating a method chain into a binding and a method call? Something like:

```rust
#[allow(clippy::expect_used)]
let __rt = #rt
    .enable_all()
    .build()
    .expect("Failed building the Runtime");
__rt.block_on(async #body)
```

Reply:
I agree that separating them seems best.

Reply:
That was my second idea before I started considering panic! or an outer result macro overhaul, but I didn't want to expose an unexpected variable to the scope of tokio user's main function / closure. If you think it's the best way to do this, I will do it :)

Reply:
> That was my second idea before I started considering panic! or an outer result macro overhaul, but I didn't want to expose an unexpected variable to the scope of tokio user's main function / closure. If you think it's the best way to do this, I will do it :)

Ok, sorry, I'm silly, the solution was quite obvious